File: backend/src/RealSync.Core/Entities/Customer.cs
namespace RealSync.Core.Entities;

/// <summary>
/// Khách hàng đã chuyển đổi từ Lead (Won status).
/// Lưu thông tin chi tiết khách hàng chính thức.
/// </summary>
public class Customer : BaseEntity
{
    public string FullName { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public string? Email { get; set; }
    public string? Address { get; set; }
    public string? Company { get; set; }
    public string? Notes { get; set; }
    public string? Source { get; set; }  // Website, Facebook, Zalo, Phone, Referral

    // Liên kết với Lead đã chuyển đổi
    public Guid? ConvertedFromLeadId { get; set; }
    public Lead? ConvertedFromLead { get; set; }

    // Người phụ trách
    public Guid? AssignedToId { get; set; }
    public User? AssignedTo { get; set; }
}

File: frontend/src/types/crm/customer.ts
import type { ApiMeta, ApiResponse } from './lead';

export type CustomerSource = 'Website' | 'Facebook' | 'Zalo' | 'Phone' | 'Referral' | 'Other';

export type CustomerActivityAction = 'Create' | 'Update' | 'Delete' | 'Assignment' | 'StatusChange' | 'View' | 'Export';

export type CustomerOriginFilter = 'all' | 'converted' | 'direct';

export interface CustomerActivityLog {
  id: string;
  userId?: string | null;
  userName?: string | null;
  entityType: 'Customer';
  entityId: string;
  action: CustomerActivityAction;
  description?: string | null;
  oldValues?: string | null;
  newValues?: string | null;
  createdAt: string;
}

export interface CrmCustomerListItem {
  id: string;
  fullName: string;
  phone?: string | null;
  email?: string | null;
  address?: string | null;
  company?: string | null;
  notes?: string | null;
  source?: CustomerSource | null;
  assignedToId?: string | null;
  assignedToName?: string | null;
  convertedFromLeadId?: string | null;
  convertedFromLeadName?: string | null;
  createdAt: string;
  updatedAt?: string | null;
  activities: CustomerActivityLog[];
}

export interface CrmCustomerDetail extends CrmCustomerListItem {}

export interface CustomerQuery {
  page: number;
  pageSize: number;
  search: string;
  source?: CustomerSource | null;
  assignedToId?: string | null;
  convertedFromLeadId?: string | null;
  convertedFromLead?: boolean | null;
  origin?: CustomerOriginFilter;
  fromDate?: string | null;
  toDate?: string | null;
  sortBy?: 'createdAt' | 'updatedAt' | 'fullName' | 'company';
  sortDirection?: 'asc' | 'desc';
}

export interface CustomerCreateModel {
  fullName: string;
  phone?: string | null;
  email?: string | null;
  address?: string | null;
  company?: string | null;
  notes?: string | null;
  source?: CustomerSource | null;
  assignedToId?: string | null;
}

export interface CustomerUpdateModel extends CustomerCreateModel {}

export interface CustomerAssignmentModel {
  assignedToId: string;
  note?: string | null;
}

export type CustomerApiResponse<T> = ApiResponse<T>;
export type CustomerApiMeta = ApiMeta;

export const CUSTOMER_SOURCES: CustomerSource[] = ['Website', 'Facebook', 'Zalo', 'Phone', 'Referral', 'Other'];
export const CUSTOMER_ACTIVITY_ACTIONS: CustomerActivityAction[] = [
  'Create',
  'Update',
  'Delete',
  'Assignment',
  'StatusChange',
  'View',
  'Export'
];
